### PR TITLE
Add note about EOL packages

### PR DESCRIPTION
**Pull request by trevor-scheer**
There's been some apparent confusion around what EOL means for AS 2/3. Concerned folks seem to be under the impression that they need to migrate away immediately as if the software is going to disappear out from under them, so we should make it clear that's not the case, even if we recommend strongly against using EOL software.

### CODE REVIEW

File: docs/source/previous-versions.mdx [hunk 1]
```diff
@@ -14,6 +14,8 @@ Both previous major versions of Apollo Server are currently [deprecated](/resour
 
 Deprecated versions continue to receive security patches and updates to address major regressions until their end-of-life date. They typically do not receive new features and other kinds of bug fixes may or may not be backported to deprecated versions at Apollo's discretion.
 
+End-of-life versions will remain available in the npm registry. They can still be installed and used in your projects, but Apollo does not guarantee any further support for them nor does Apollo recommend their use.
```
Comment (glasser) [inline docs/source/previous-versions.mdx:17]: maybe something about how other systems might change in ways that they don't work any more, like Studio may get changed in ways that Studio integration features from EOL versions don't work?

Comment (maryelizbeth) [reply docs/source/previous-versions.mdx:17]: took a stab at this: 

```EOL versions can still be used in your projects, but may no longer function as expected when used with Apollo Studio or GraphOS beyond their end-of-life date. Apollo does not guarantee support for or recommend the use of EOL versions```  

feel free to use or discard any bits that work

Comment (trevor-scheer) [reply docs/source/previous-versions.mdx]: I reworked it before I saw your comment. I should mention GraphOS but let me know what you think of the update @maryelizbeth

File: docs/source/previous-versions.mdx [hunk 2]
```diff
@@ -14,6 +14,8 @@ Both previous major versions of Apollo Server are currently [deprecated](/resour
 
 Deprecated versions continue to receive security patches and updates to address major regressions until their end-of-life date. They typically do not receive new features and other kinds of bug fixes may or may not be backported to deprecated versions at Apollo's discretion.
 
+End-of-life versions will remain available in the npm registry. They can still be installed and used in your projects, but Apollo does not guarantee any further support for them nor does Apollo recommend their use. Apollo makes no guarantee that Apollo Studio or GraphOS features will continue to work as expected in tandem with packages that are beyond their end-of-life date.
```
Comment (maryelizbeth) [inline docs/source/previous-versions.mdx:17]: 👍 LGTM

**Review by maryelizbeth — APPROVED** [review]

File: docs/source/previous-versions.mdx (hunk 2 above)
Comment (rkoron007) [inline docs/source/previous-versions.mdx:17]: Slight suggested tweaks for style, but beyond that looks great to me!
```suggestion
End-of-life versions will remain available in the npm registry and can still be installed and used in your projects. However, Apollo doesn't guarantee any further support for them, nor does Apollo recommend their use. Additionally, Apollo doesn't guarantee that GraphOS features will continue to work for packages beyond their end-of-life date.
```

**Review by rkoron007 — COMMENTED** [review]
Looks good with some slight tweaks for style!

**Pull request merged**